proxy/src/control/discovery.rs

```diff
@@ -404,12 +370,11 @@ where
                             set.txs.push(tx);
                         }
                         Entry::Vacant(vac) => {
-                            let query =
-                                DestinationServiceQuery::connect_maybe(
-                                    &self.default_destination_namespace,
-                                    client,
-                                    vac.key(),
-                                    "connect");
+                            let query = Self::query_destination_service_if_valid_auth(
```

Comment:
`valid_auth` seems like a poor name to me. Really the distinction isn't "valid" vs "invalid" but rather "may be an internal service" vs "is definitely not an internal service". I suggest `query_destination_service_if_relevant` or similar. In particular I would like to avoid implying anything about the validity of the authority in the name.